File: benchmarks/reference/fib.java

public class fib {
    static long fib(int n) {
        if (n < 2) return n;
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        long start = System.nanoTime();
        
        System.out.println("Fibonacci(30): " + fib(30));
        
        long end = System.nanoTime();
        double elapsed = (end - start) / 1_000_000_000.0;
        
        System.out.println("Time: " + elapsed);
    }
}

File: benchmarks/reference/fib.js

function fib(n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

const start = process.hrtime();
console.log("Fibonacci(30): " + fib(30));
const diff = process.hrtime(start);
const elapsed = diff[0] + diff[1] / 1e9;
console.log("Time: " + elapsed.toFixed(6));
